gems/manageiq-providers-vmware-vcd/app/models/manageiq/providers/vmware_vcd/cloud_manager/refresher.rb

```diff
@@ -0,0 +1,71 @@
+class ManageIQ::Providers::VmwareVcd::CloudManager::Refresher < ManageIQ::Providers::BaseManager::Refresher
+  include ::EmsRefresh::Refreshers::EmsRefresherMixin
+
+  def collect_inventory_for_targets(ems, targets)
+    vcd = ems.vcd_service
+
+    inv = {
+      :orgs  => [],
+      :vdcs  => [],
+      :vapps => [],
+      :vms   => [],
+    }
+
+    vcd.organizations.all.each do |org|
+      inv[:orgs].push(org)
+    end
+
+    inv[:orgs].each do |org|
+      org.vdcs.all.each do |vdc|
+        inv[:vdcs].push(vdc)
+      end
```

Comment:
What is there the need to loop and push?  If .all returns a collection it should be directly dumpable into the array (perhaps a `.to_a` call or something?)

```ruby
inv[:vdcs] = org.vdcs.all
# or
inv[:vdcs] = org.vdcs.all.to_a
# or
inv[:vdcs] = org.vdcs.all.dup
```